frontend/src/stores/form.js

```diff
@@ -0,0 +1,34 @@
+import { defineStore } from 'pinia';
+import { objectDeepCopy } from '@/helpers/object';
+import { useStorage } from '@vueuse/core';
+
+const defaultSubjectFormState = {
+  name: '',
+  timesPerWeek: 0,
+  teacher: { name: '' },
+  classroom: { name: '' },
+};
+
+const defaultClassFormState = {
+  name: '',
+  subjects: [defaultSubjectFormState],
+};
+
+const defaultFormState = {
+  classes: [defaultClassFormState],
+};
+
+export const useFormStore = defineStore('form', () => {
+  const form = useStorage('form', objectDeepCopy(defaultFormState));
```

Comment:
Is there even a need for deep cloning? Does `useStorage` return a ref? If it does, it should work the same way as `reactive` where the [returned object is not the as the passed one](https://vuejs.org/guide/essentials/reactivity-fundamentals.html#reactive-proxy-vs-original-1).

Reply:
yes, there is need because new instances of class or subject that are pushed to array will have old data in it. I think this is problem when there are nested objects because of some references or something, but this is definitely a JS thing and not a vue thing because I've encountered same problem in react

Reply:
Fair enough. I assumed such things were resolved in Vue 3 and their updated reactivity approach. Leave it as is and I will play around with that at some point in the future.